oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -663,12 +686,16 @@ static final class ServiceAccountImpersonationOptions implements java.io.Seriali
 
     private final int lifetime;
 
+    private final boolean customLifetimeProvided;
+
     ServiceAccountImpersonationOptions(Map<String, Object> optionsMap) {
       if (!optionsMap.containsKey(TOKEN_LIFETIME_SECONDS_KEY)) {
         lifetime = DEFAULT_TOKEN_LIFETIME_SECONDS;
+        customLifetimeProvided = false;
```

Comment:
this can be changed to:
```
customTokenLifetimeRequested = optionsMap.containsKey(TOKEN_LIFETIME_SECONDS_KEY);
if (!customTokenLifetimeRequested) {
  lifetime = DEFAULT..
}
```